src/zope/app/apidoc/utilities.py

```diff
@@ -41,11 +42,20 @@
 _remove_html_overhead = re.compile(
     r'(?sm)^<html.*<body.*?>\n(.*)</body>\n</html>\n')
 
-space_re = re.compile('\n^( *)\S', re.M)
+space_re = re.compile(r'\n^( *)\S', re.M)
 
 _marker = object()
 
-BASEDIR = dirname(dirname(dirname(dirname(zope.app.__file__))))
+try:
+    _path = zope.app.__file__
+except AttributeError:
+    try:
+        _path = zope.app.__path__[0]
+    except TypeError:
+        # "_NamespacePath" object does not support indexing
+        _path = list(zope.app.__path__)[0]
+BASEDIR = dirname(dirname(dirname(dirname(_path))))
```

Comment:
So this goes from `random-venv-path/lib/pythonX.Y/site-packages/zope/app/__init__.py` to `random-venv-path/lib/pythonX.Y` because ... ?

What if we're using zc.buildout and zope.app is in `~/.cache/buildout/eggs/zope.app-X.Y.Z.egg/zope/app/`?  Does it make sense to have BASEDIR pointing to `~/.cache/buildout/eggs`?

Instead of making this code work on Python 3 can we delete it?  What is BASEDIR used for?  I've a suspicion it assumes the old monolithic Zope 3 source tree we had pre-eggsplosion.